webapp/src/Controller/API/PrintController.php

```diff
@@ -0,0 +1,143 @@
+            properties: [
+                new OA\Property(
+                    property: "original_name",
+                    description: "The original name of the file",
+                    type: "string"
+                ),
+                new OA\Property(
+                    property: "language",
+                    description: "The programming language of the file contents",
+                    type: "string"
+                ),
+                new OA\Property(
+                    property: "file_contents",
+                    description: "The plaintext contents of the source file",
+                    type: "string"
+                ),
+            ],
+            type: "object"
+        )
+    )]
+    #[OA\Response(
+        response: 200,
+        description: "Returns the ID of the imported problem and any messages produced",
+        content: new OA\JsonContent(
+            properties: [
+                new OA\Property(
+                    property: "success",
+                    description: "Whether printing was successful",
+                    type: "boolean"
+                ),
+                new OA\Property(
+                    property: "output",
+                    description: "The print command output",
+                    type: "string"
+                ),
+            ],
+            type: "object"
+        )
+    )]
+    #[OA\Response(
+        response: 422,
+        description: "An error occurred while processing the print job",
+        content: new OA\JsonContent(
+            properties: [
+                new OA\Property(property: "errors", type: "object"),
+            ]
+        )
+    )]
+    public function printAsTeam(Request $request): JsonResponse
+    {
+        $origname = $request->request->get("original_name");
+        $contents = $request->request->get("file_contents");
+        /** @var array<string,string> error messages */
+        $errors = [];
+
+        $language = $request->request->get("language");
+        $langid = $this->em
+            ->createQueryBuilder()
+            ->from(Language::class, "l")
+            ->select("l.langid")
+            ->andWhere("l.allowSubmit = 1")
+            ->andWhere("l.name = :name")
+            ->setParameter("name", $language)
+            ->getQuery()
+            ->getScalarResult();
+        if (empty($langid)) {
+            $errors[] = "Programming language not found";
+        }
+
+        if (!($tempFilename = tempnam($this->dj->getDomjudgeTmpDir(), "printfile-"))) {
+            throw new ServiceUnavailableHttpException(null, "Could not create temporary file.");
+        }
+
+        if (file_put_contents($tempFilename, $contents) === false) {
+            throw new ServiceUnavailableHttpException(
+                null,
+                sprintf("Could not write printfile to temporary file '%s'.", $tempFilename)
+            );
+        }
+
+        // Handle validation errors
+        if (!empty($errors)) {
```

Comment:
There can only be one error it seems (line 108). Should we just return there?

Reply:
We could do that. I felt like following the same structure as other places might be nice.